### ISSUE TITLE
JAWS 2023: Not announcing dialog when inner button is focused upon opening

### ISSUE THREAD

**Issue by adamsamec** [FreedomScientific/VFO-standards-support#684]
### Summary

Whenever a dialog is displayed and keyboard focus is programatically set on a button within that dialog upon dialog opening, JAWS 2023 does not announce that dialog has been entered. This bug occurs only when a button and maybe other type of element is focused upon dialog opening, as, for instance, when an input text field is focused instead of a button, the entrance into dialog is announced correctly.

#### Steps to reproduce

1. Turn on JAWS 2023.
2. Visit the [example page](https://jsbin.com/konopipuwo/edit?html,output).
3. Activate the "Open dialog #1 example" button.
4. Close the dialog using the "Close" or "Got it" button.
5. Activate the "Open dialog #2 example" button.

### Expected result

After the repro step number 3, when the dialog is being displayed and focus is moved to an inner button, JAWS should announce that dialog is opening along with its accessible name.

### Actual result

Only accessible name of the dialog is announced, that is, without mentioning that dialog is opening.

Compare the announcement with activating the button named "Open dialog #2 example" as instructed in the repro step number 5, where the difference is that, upon dialog opening, an input text field is focused instead of a button. In this case, the announcement correctly includes that a dialog is opening.

### Example

[Visit https://jsbin.com/konopipuwo/edit?html,output](https://jsbin.com/konopipuwo/edit?html,output).
        
#### JAWS version and build number

 JAWS 2023.2210.29.
 
#### Operating System and version

Microsoft Windows 11, version 22H2 (OS Build 22621.819).

#### Browser and version:

The issue has been tested and observed in Google Chrome 107, Microsoft Edge 107 and Firefox 107, but it probably occurs in other web browsers as well.
